### PR TITLE
Add Teleport to vendor-technical-contacts folder

### PR DESCRIPTION
**Pull request by strideynet**
As per https://github.com/spiffe/spiffe/blob/main/community/vendor-companies-engagement-procedure.md

Perhaps we need a template for these? I've tried a rough structure here but I'm open to feedback :)

### CODE REVIEW

File: community/vendor-technical-contacts/teleport.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Support
+
+Enterprise customers should seek support via https://goteleport.com/support/
+
+OSS users can should seek support via the community Slack: https://goteleport.com/community-slack/
+
+## Contacts
+
+### Noah Stride (Technical && Community)
```
Comment (heymarcel) [inline community/vendor-technical-contacts/teleport.md]: Typo: Double ampersand (unless you want a logical AND) :)

Comment (strideynet) [reply community/vendor-technical-contacts/teleport.md]: It's possible I'd spent a few too many hours coding before writing this 😭

Comment (strideynet) [reply community/vendor-technical-contacts/teleport.md]: [57431e3](https://github.com/spiffe/spiffe/pull/298/commits/57431e33f63a2d2f736fedfcdd8ba884876bf8ef)

**Comment by amoore877** [comment]
don't forget the DCO check :)

> Perhaps we need a template for these?

@v0lkan thoughts?

**Comment by v0lkan** [comment]
> don't forget the DCO check :)
> 
> > Perhaps we need a template for these?
> 
> @v0lkan thoughts?

It might be worthwhile to have a bunch of these to decide a common template. 
Similar to reusing the same code piece three times before exporting it to a shared functions
(sorry; I have been coding a lot nowadays as well; hence the analogy)

How different can you write contact information anyway :). 

But if we need a dozen of such fields, then a template can be helpful to standardize things.

**Comment by strideynet** [comment]
> don't forget the DCO check :)

Should be signed off now 👍

**Review by amoore877 — APPROVED** [review]

**Pull request merged**